src/App.js

```diff
@@ -676,6 +671,48 @@ function FullApp() {
     };
   }, [active, chainId, vaultAddress, positionRouterAddress]);
 
+  const { data: tokenBalances } = useSWR(
+    [`FullApp:getTokenBalances:${active}`, chainId, readerAddress, "getTokenBalances", account || PLACEHOLDER_ACCOUNT],
+    {
+      fetcher: fetcher(library, ReaderV2, [tokenAddresses]),
+    }
+  );
+  const { data: fundingRateInfo } = useSWR([active, chainId, readerAddress, "getFundingRates"], {
+    fetcher: fetcher(library, ReaderV2, [vaultAddress, nativeTokenAddress, whitelistedTokenAddresses]),
+  });
+
+  const { infoTokens } = useInfoTokens(library, chainId, active, tokenBalances, fundingRateInfo);
+
+  // Track user wallet connect
+  useEffect(() => {
+    const accountChanged = hasChangedAccount(account);
+    if ((!loggedInTracked || accountChanged) && infoTokens) {
+      const sendTrackLoginData = async () => {
+        const MAX_DECIMALS = 16;
```

Comment:
Changed to 18

Reply:
they might not always be 18, unless getBalanceAndSupplyData formats it. But I thought it just called getBalance. In which case it will whatever token decimals that token is